src/components/SingleRecipePage.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState, useEffect } from "react";
+import {
+  Box,
+  Collapse,
+  Container,
+  Icon,
+  IconButton,
+  Image,
+  Flex,
+  Grid,
+  GridItem,
+  Heading,
+  Text,
+  useDisclosure
+} from "@chakra-ui/react";
+import { useParams } from "react-router-dom";
+import { SavedRecipe } from "../utils/types";
+import { getSingleRecipe } from "../utils/fetchData";
+import { ArrowBackIcon, ChevronDownIcon } from "@chakra-ui/icons";
+import { GiPencil, GiCalendar, GiShoppingCart } from "react-icons/gi";
+import { IoTrashOutline } from "react-icons/io5";
+import { TfiPrinter } from "react-icons/tfi";
+import SingleRecipeIngredients from "./SingleRecipeIngredients";
+import SingleRecipeTag from "./SingleRecipeTag";
+
+const SingleRecipePage = () => {
+  const [recipe, setRecipe] = useState<SavedRecipe>();
+  const { slug } = useParams();
+  const id = slug;
+  const { isOpen, onToggle } = useDisclosure();
+  useEffect(() => {
+    if (id === undefined) return;
+    getSingleRecipe(id)
+      .then(response => {
+        console.log(response.data);
+        setRecipe(response.data);
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  }, [id]);
+  if (recipe === undefined) return;
+
+  return (
+    <Container maxW="5xl">
+      <Grid templateColumns="repeat(3, 1fr)" gap={6}>
+        <GridItem colSpan={2} w="100%">
+          <Flex marginTop="10" marginBottom="5" alignItems="center" gap={2}>
+            <IconButton
+              size="lg"
+              variant="ghost"
+              aria-label="on the previous page"
+              icon={<ArrowBackIcon />}
+              title="on the previous page"
+            />
+            <Heading size="lg">{recipe.recipeName.toUpperCase()}</Heading>
+          </Flex>
+          <Box>
+            <Text as="span">
+              <b>Prep time:</b>&nbsp;
+              {`${recipe.recipePrepTime.recipePrepTimeMinutes} min`}
+              &nbsp;&nbsp;
+            </Text>
+            <Text as="span">
+              <b>Cooking time:</b>&nbsp;
+              {`${recipe.recipeCookTime.recipeCookTimeMinutes} min`}
+              &nbsp;&nbsp;
+            </Text>
+            <Text as="span">
+              <b>Total:</b>&nbsp;
+              {`${
+                recipe.recipePrepTime.recipePrepTimeMinutes +
+                recipe.recipeCookTime.recipeCookTimeMinutes
+              } min`}
+              &nbsp;&nbsp;
+            </Text>
+          </Box>
+          <Text>
+            <b>Servings:</b>&nbsp;
+            {`${recipe.recipeServings}`}
+            &nbsp;&nbsp;
+          </Text>
+        </GridItem>
+        <GridItem colSpan={1} w="100%" position="relative">
+          <Flex
+            w="100%"
+            gap="2"
+            flexShrink="1"
+            flexWrap="wrap"
+            position="absolute"
+            bottom="2"
+            // alignSelf=""
```

Comment:
NON-BLOCKING: Do you still need this?

Reply:
Thanks! I missed it! I'll remove in the branch I am working right now.